pynumdiff/total_variation_regularization/_total_variation_regularization.py

```diff
@@ -1,123 +1,112 @@
+            if 'cg_maxiter' in options: cg_maxiter = options['cg_maxiter']
+            if 'scale' in options: scale = options['scale']
+    elif num_iterations == None or gamma == None:
+        raise ValueError("`num_iterations` and `gamma` must be given.")
+
+    dxdt_hat = _chartrand_tvregdiff.TVRegDiff(x, num_iterations, gamma, dx=dt,
+                                                maxit=cg_maxiter, scale=scale,
+                                                ep=1e-6, u0=None, plotflag=False, diagflag=1)
     x_hat = utility.integrate_dxdt_hat(dxdt_hat, dt)
     x0 = utility.estimate_initial_condition(x, x_hat)
     x_hat = x_hat + x0
 
     return x_hat, dxdt_hat
 
 
-# Generalized total variation regularized derivatives
-def __total_variation_regularized_derivative__(x, dt, N, gamma, solver='MOSEK'):
-    """
-    Use convex optimization (cvxpy) to solve for the Nth total variation regularized derivative.
-    Default solver is MOSEK: https://www.mosek.com/
-
-    :param x: (np.array of floats, 1xN) time series to differentiate
-    :param dt: (float) time step
-    :param N: (int) 1, 2, or 3, the Nth derivative to regularize
-    :param gamma: (float) regularization parameter
-    :param solver: (string) Solver to use. Solver options include: 'MOSEK' and 'CVXOPT',
-                            in testing, 'MOSEK' was the most robust.
-    :return: x_hat    : estimated (smoothed) x
-             dxdt_hat : estimated derivative of x
+def _total_variation_regularized_derivative(x, dt, order, gamma, solver=solver):
+    """Generalized total variation regularized derivatives. Use convex optimization (cvxpy) to solve for a
+    total variation regularized derivative. Default solver is MOSEK: , which is not
+    always available. Fall back to CVXPY's default.
+
+    :param np.array[float] x: data to differentiate
+    :param float dt: time step
+    :param int order: 1, 2, or 3, the derivative to regularize
+    :param float gamma: regularization parameter
+    :param str solver: Solver to use. Solver options include: 'MOSEK', 'CVXOPT', 'CLARABEL', 'ECOS'.
+                            In testing, 'MOSEK' was the most robust.
+  
+    :return: tuple[np.array, np.array] of\n
+             - **x_hat** -- estimated (smoothed) x
+             - **dxdt_hat** -- estimated derivative of x
     """
-
     # Normalize
     mean = np.mean(x)
     std = np.std(x)
     x = (x-mean)/std
 
     # Define the variables for the highest order derivative and the integration constants
-    var = cvxpy.Variable(len(x) + N)
+    deriv_values = cvxpy.Variable(len(x)) # values of the order^th derivative, in which we're penalizing variation
+    integration_constants = cvxpy.Variable(order) # constants of integration that help get us back to x
 
     # Recursively integrate the highest order derivative to get back to the position
-    derivatives = [var[N:]]
-    for i in range(N):
-        d = cvxpy.cumsum(derivatives[-1]) + var[i]
-        derivatives.append(d)
```

Comment:
You were storing all the derivatives before, which I don't think is necessary, since we only use the final construction of these values. I now don't bother remembering all the levels.